core/services/ardupilot_manager/ArduPilotManager.py

```diff
@@ -188,7 +201,28 @@ def start_board(self, boards: List[Tuple[FlightControllerType, str]]) -> bool:
         raise RuntimeError("Invalid board type: {boards}")
 
     def restart(self) -> None:
-        self.mavlink_manager.restart()
+        self.mavlink_manager.stop()
+        if self.current_platform == Platform.SITL:
+            self.run_with_sitl(self.current_sitl_vehicle)
+            return
+        self.run_with_board()
+
+    def prune_ardupilot_processes(self) -> None:
+        if self.subprocess:
+            logger.debug("Terminating main Ardupilot process.")
+            self.subprocess.kill()
+            self.subprocess.wait(5)
+
+        def is_ardupilot_process(process: psutil.Process):
+            return str(self.firmware_installer.firmware_path) in " ".join(process.cmdline())
```

Comment:
Can you explain or add a comment about it ? This line is not clear with the function name.